src/app/tasks/components/TaskLogger/TaskLogger.tsx

```diff
@@ -38,18 +39,29 @@ const TaskLogger = (): JSX.Element => {
       }, {}),
     [filesToRetry],
   );
-  const items: JSX.Element[] = allNotifications.map((n) => (
-    <TaskLoggerItem
-      notification={n}
-      task={tasksService.findTask(n.taskId)}
-      key={n.taskId}
-      filesToRetry={filesToRetryGroupedByTask[n.taskId]}
-    />
-  ));
+
+  const refreshItems = () => {
+    setItems(
+      allNotifications.map((n) => (
+        <TaskLoggerItem
+          notification={n}
+          task={tasksService.findTask(n.taskId)}
+          key={n.taskId}
+          filesToRetry={filesToRetryGroupedByTask[n.taskId]}
+        />
+      )),
+    );
+  };
+
+  useEffect(() => {
+    refreshItems();
+  }, [allNotifications, filesToRetryGroupedByTask]);
```

Comment:
nit: I think this could be replaced by using memoization instead the useEffect:
```
const items = useMemo(() => {
  return allNotifications.map((n) => (
    <TaskLoggerItem
      notification={n}
      task={tasksService.findTask(n.taskId)}
      key={n.taskId}
      filesToRetry={filesToRetryGroupedByTask[n.taskId]}
    />
  ));
}, [allNotifications, filesToRetryGroupedByTask]);
```